build.gradle

```diff
@@ -119,3 +120,10 @@ compileJava {
 compileTestJava {
     options.compilerArgs += ["-Xlint:unchecked", "-Xlint:deprecation"]
 }
+
+startScripts {
+    doLast {
+        unixScript.text = unixScript.text
+            .replace('DEFAULT_JVM_OPTS=\"\"', 'DEFAULT_JVM_OPTS=\"-Xmx10g\"')
```

Comment:
The behavior is the same as on the server - the JVM will be initialized with a maximum of 10G of heap space. That just means it will not grow beyond the 10G, so I don't think this should be a problem on the laptop.